libraries/net/node.cpp

```diff
@@ -5051,4 +5064,70 @@ namespace graphene { namespace net { namespace detail {
 
   } // end namespace detail
 
+   /***
+    * @brief Helper to convert a string to a collection of endpoints
+    *
+    * This converts a string (i.e. "bitshares.eu:665535" to a collection of endpoints.
+    * NOTE: Throws an exception if not in correct format or was unable to resolve URL.
+    *
+    * @param in the incoming string
+    * @returns a vector of endpoints
+   */
+   std::vector<fc::ip::endpoint> node::resolve_string_to_ip_endpoints(const std::string& in)
+  {
+      try
+      {
+         std::string::size_type colon_pos = in.find(':');
+         if (colon_pos == std::string::npos)
+            FC_THROW("Missing required port number in endpoint string \"${endpoint_string}\"",
+                  ("endpoint_string", in));
+         std::string port_string = in.substr(colon_pos + 1);
+         try
+         {
+            uint16_t port = boost::lexical_cast<uint16_t>(port_string);
+
+            std::string hostname = in.substr(0, colon_pos);
+            std::vector<fc::ip::endpoint> endpoints = fc::resolve(hostname, port);
+            if (endpoints.empty())
+               FC_THROW_EXCEPTION( fc::unknown_host_exception,
+                     "The host name can not be resolved: ${hostname}",
+                     ("hostname", hostname) );
+            return endpoints;
+         }
+         catch (const boost::bad_lexical_cast&)
+         {
+            FC_THROW("Bad port: ${port}", ("port", port_string));
+         }
+      }
+      FC_CAPTURE_AND_RETHROW((in))
+   }
+
+   void node::add_seed_node(const std::string& endpoint_string)
+   {
+      INVOKE_IN_IMPL(add_seed_node, endpoint_string);
+   }
+
+   /*****
+    * @brief add a list of nodes to seed the p2p network
+    * @param seeds a vector of url strings
+    * @param connect_immediately attempt a connection immediately
+    */
+   void node::add_seed_nodes(std::vector<std::string> seeds)
+   {
+      for(const std::string& endpoint_string : seeds )
+      {
+         try {
+            INVOKE_IN_IMPL(add_seed_node, endpoint_string);
```

Comment:
One seed: I do not believe that to be the case, but will build a test.

Exception in other "thread": I will also investigate.

Reply:
@abitmore is correct. Instead of using `INVOKE_IN_IMPL` here, you should call `node::add_seed_node`, then *that* will use `INVOKE_IN_IMPL` and return the resulting future.

Since the future is discarded, exceptions from `add_seed_node` will be ignored. This is the same as the current behaviour.

Reply:
I added node::add_seed_node. See https://github.com/bitshares/bitshares-core/pull/1764/commits/1b8269ba3cfb47ef5606b144fe1c36f3ae95fc8a

With the use of INVOKE_IN_IMPL, I believe that I no longer need the try/catch in node::add_seed_nodes. Please verify and I will remove it.

Reply:
Note:
Besides of "failed-to-resolve-an-address", the old code would trigger an `already_connected_to_requested_peer` exception. It's fixed in this PR due to removal of this line: https://github.com/bitshares/bitshares-core/pull/1764/files#diff-96eeb53aa89c9bb1313e838c970913f0L136.